File: sol/solution/0301-0400/0303/Solution.kt
class NumArray(var nums: IntArray) {
    var prefix  = 0
    init{
        for(i in nums.indices){
            if(i == 0)
            prefix = nums[i]
            else{
                nums[i] = prefix + nums[i]
            prefix = nums[i]
            }
        }
    }
    fun sumRange(left: Int, right: Int): Int {
        if(left == 0){
            return nums[right]
        }else
        return nums[right] - nums[left - 1]
    }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * var obj = NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */

File: sol/solution/0301-0400/0303/Solution.php
class NumArray {
    /**
     * @param Integer[] $nums
     */
    private $prefixSum;
    function __construct($nums) {
        $this->prefixSum = [];
        $sum = 0;
        foreach ($nums as $num) {
            $sum += $num;
            $this->prefixSum[] = $sum;
        }
    }
    /**
     * @param Integer $left
     * @param Integer $right
     * @return Integer
     */
    function sumRange($left, $right) {
        if ($left == 0) {
            return $this->prefixSum[$right];
        }
        return $this->prefixSum[$right] - $this->prefixSum[$left - 1];
    }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * $obj = NumArray($nums);
 * $ret_1 = $obj->sumRange($left, $right);
 */